elements/lisk-p2p/src/p2p.ts

```diff
@@ -541,10 +548,19 @@ export class P2P extends EventEmitter {
 	 * invoke an async RPC on Peers to give them our new node status.
 	 */
 	public applyNodeInfo(nodeInfo: P2PNodeInfo): void {
-		this._nodeInfo = {
-			...nodeInfo,
-		};
-		this._peerPool.applyNodeInfo(this._nodeInfo);
+		try {
+			const maxPeerInfoSize = this._config.maxPeerInfoSize
+				? this._config.maxPeerInfoSize
+				: DEFAULT_MAX_PEER_INFO_SIZE;
+
+			this._nodeInfo = {
+				...validateNodeInfo(nodeInfo, maxPeerInfoSize),
```

Comment:
I think destructuring `validateNodeInfo` doesn't make much sense.
maybe call in the different line?